openvpn/pia/update-port.sh

```diff
@@ -130,44 +123,35 @@ if [[ "$new_port" != "$transmission_peer_port" ]]; then
 else
     echo "No action needed, port hasn't changed"
 fi
-
-echo ""
-echo "initial setup complete!"
-echo ""
-echo "waiting for rebind loop................."
-
-echo "token expiry $pf_token_expiry"
-pf_remaining=$((  $pf_token_expiry - $(date +%s) ))
-echo "remaining = $pf_remaining"
-pf_bindinterval=$(( 30 * 60))
-# Get a new token when the current one has less than this remaining
-# Defaults to 7 days (same as desktop app)
+}
+echo "Running functions for token based port fowarding"
+get_auth_token
+get_sig
+bind_port
+bind_trans
+format_expiry=$(date -d @$pf_token_expiry)
+echo "#######################"
+echo "        SUCCESS        "
+echo "#######################"
+echo "Port: $pf_port"
+echo "Expiration $format_expiry"
+echo "#######################"
+echo "Entering infinite while loop"
+echo "Every 15 minutes, check port status"
 pf_minreuse=$(( 60 * 60 * 24 * 7 ))
-
-pf_remaining=0
-pf_firstrun=1
-vpn_ip=$(ip route | grep tun | grep -v src | head -1 | awk '{ print $3 }')
-pf_host="$vpn_ip"
+pf_remaining=$((  $pf_token_expiry - $(date +%s) ))
 
 while true; do
-  pf_remaining=$((  $pf_token_expiry - $(date +%s) ))
-  # Get a new pf token as the previous one will expire soon
-  if [ $pf_remaining -lt $pf_minreuse ]; then
-    if [ $pf_firstrun -ne 1 ]; then
-      echo "$(date): PF token will expire soon. Getting new one."
-    else
-      echo "$(date): Getting PF token"
-      pf_firstrun=0
-    fi
-    get_sig
-    echo "$(date): Obtained PF token. Expires at $pf_token_expiry_raw"
-    bind_port
-    echo "$(date): Server accepted PF bind"
-    echo "$(date): Forwarding on port $pf_port"
-    echo "$(date): Rebind interval: $pf_bindinterval seconds"
-  fi
-  sleep $pf_bindinterval &
-  wait $!
-  
-bind_port  
+	pf_remaining=$((  $pf_token_expiry - $(date +%s) ))
```

Comment:
Remaining =[ (static value always is 60ish depending on PIA)  - date in ms] test: Its Dec 5. Got port Dec 1. Foot until Feb 1. This script returns 56 days~ depending on token expiry. In a week it will be 49